File: src/Atlasd/Battlenet/Exceptions/ProtocolNotSupportedException.cs
using System;

namespace Atlasd.Battlenet.Exceptions
{
    class ProtocolNotSupportedException : ClientException
    {
        public ProtocolType.Types ProtocolType { get; private set; }

        public ProtocolNotSupportedException(ProtocolType.Types protocolType, ClientState client, string message = "Unsupported protocol") : base(client, message)
        {
            ProtocolType = protocolType;
        }

        public ProtocolNotSupportedException(ProtocolType.Types protocolType, ClientState client, string message, Exception innerException) : base(client, message, innerException)
        {
            ProtocolType = protocolType;
        }
    }
}

File: src/AtlasV/Battlenet/Exceptions/ProtocolNotSupportedException.vb
Imports System

Namespace AtlasV.Battlenet.Exceptions
    Class ProtocolNotSupportedException
        Inherits ClientException

        Public Property ProtocolType As ProtocolType.Types

        Public Sub New(ByVal varProtocolType As ProtocolType.Types, ByVal varClient As ClientState, ByVal Optional varMessage As String = "Unsupported protocol")
            MyBase.New(varClient, varMessage)
            ProtocolType = varProtocolType
        End Sub

        Public Sub New(ByVal varProtocolType As ProtocolType.Types, ByVal varClient As ClientState, ByVal varMessage As String, ByVal varInnerException As Exception)
            MyBase.New(varClient, varMessage, varInnerException)
            ProtocolType = varProtocolType
        End Sub
    End Class
End Namespace
